compiler/rustc_data_structures/src/sync.rs

```diff
@@ -82,6 +82,14 @@ cfg_if! {
             }
         }
 
+        impl Atomic<bool> {
+            pub fn fetch_or(&self, val: bool, _: Ordering) -> bool {
+                let result = self.0.get() | val;
+                self.0.set(val);
+                result
+            }
```

Comment:
I think this is not equivalent of AtomicBools' fetch_or. Perhaps we want the following?

```suggestion
            pub fn fetch_or(&self, val: bool, _: Ordering) -> bool {
                let old = self.0.get();
                self.0.set(old | val);
                old
            }
```